sdk/attestation/Azure.Security.Attestation/src/AttestationToken.cs

```diff
@@ -119,6 +119,7 @@ protected AttestationToken()
         public virtual bool ValidateToken(IReadOnlyList<AttestationSigner> attestationSigningCertificates, Func<AttestationToken, AttestationSigner, bool> validationCallback = default)
         {
             Argument.AssertNotNull(attestationSigningCertificates, nameof(attestationSigningCertificates));
+            Argument.AssertNotNull(attestationSigningCertificates[0], nameof(attestationSigningCertificates));
```

Comment:
This will still throw an IndexOutOfRangeException if the array is empty.

Reply:
Aaah, fudge. You're right. Well, I guess it's just as well that I added the resx file so I can provide a better exception here (there's no Argument.Assert API which works for "this must be true").

Reply:
There is Argument.AssertNotNullOrEmpty

Reply:
Ah - I think I can use Argument.IsInRange for this and it'll work :).

Reply:
Oh, I assumed that AssertNotNullOrEmpty applied only to string values, not array values.